src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -1723,8 +1725,15 @@ UniqueJobId(void)
 	jobIdDatum = DirectFunctionCall1(nextval_oid, sequenceIdDatum);
 	jobId = DatumGetInt64(jobIdDatum);
 
+	/*
+	 * Add the local group id information to the jobId to
+	 * prevent concurrent jobs on different groups to conflict.
+	 */
+	localizedJobId = jobId | (localGroupId << 32);
```

Comment:
Not that it's that likely - but it's conceivable to extend 32bit in the jobid sequence.  Perhaps we should cast jobid to uint32 and simply % PG_UINT32_MAX jobid beforehand? It's really unlikely to have jobs from before / after 4 billion.    A test with such big ids won't hurt either...